### DISCUSSION TITLE
Is VS displaying for internal names expected?

### DISCUSSION THREAD

**Discussion by kyoyama-kazusa** [dotnet/roslyn#69808] [Ideas]
Hey.

I wrote a very-long query expression like this:

![image](https://github.com/dotnet/roslyn/assets/23616315/15fc19d6-1091-4ba6-8cac-3bb36292b4d3)

```csharp
var result = (ReadOnlySpan<string>)[puzzle];
while (result is [var r, ..] && r.IndexOf('.') != -1)
{
    result = (
        from solution in result
        let index = solution.IndexOf('.')
        let column = index % 9
        let block = index - index % 27 + column - index % 3
        from digit in DigitCharacters
        let duplicateCases =
            from pos in Digits
            let duplicatesInRow = solution[index - column + pos] == digit
            let duplicatesInColumn = solution[column + pos * 9] == digit
            let duplicatesInBlock = solution[block + pos % 3 + pos / 3 * 9] == digit
            where duplicatesInRow || duplicatesInColumn || duplicatesInBlock
            select pos
        where duplicateCases.Length == 0
        select $"{solution[..index]}{digit}{solution[(index + 1)..]}"
    ).ToArray();
}

return result;
```

When I put cursor onto the second `let` keyword (`let column = ...`), VS gives me an anonymous type reference, which is correct.

![image](https://github.com/dotnet/roslyn/assets/23616315/4ccb719f-3f2d-4a25-940a-a8e928e19347)

But... It seems weird. The name `<>h__TransparentIdentifier0` is too weird to be displayed. I want to know whether this is expected or not.

In my personally opinion, here the transparent identifiers like below should not be displayed. Instead, the anonymous types can be displayed in a normal way:

```csharp
new { new { string solution, int index }, int column }
```

instead of

```txt
<anonymous type: <anonymous type: string solution, int index> <>h__TransparentIdentifier0, int column>
```

And here :)

![image](https://github.com/dotnet/roslyn/assets/23616315/3d4893fb-2f2c-4237-9629-5b23defc24be)

too long... Can the displaying for nested anonymous types be shorter?

Thanks!

**Comment by jasonmalinowski** [comment]
Ah that's a bug, Please file it! :smile: FYI to @CyrusNajmabadi.

**Reply by kyoyama-kazusa** [reply]
Here is the code snippet to reproduce:

```csharp
using System;
using System.Collections.Generic;
using System.Linq;

_ =
    from element in new List<int> { 1, 2, 3 }
    let elementInterim = element + 42
    let anotherElementInterim = elementInterim - 42 // Point to 'let' keyword
    select element;


file static class Extensions
{
    /// <summary>
    /// Gets a list of <typeparamref name="TResult"/> elements.
    /// </summary>
    public static List<TResult> Select<T, TResult>(
        this List<T> elements,
        Func<T, TResult> selector
    )
    {
        var result = new List<TResult>(elements.Count);
        foreach (var element in elements)
            result.Add(selector(element));
        return result;
    }
}
```

[SharpLab](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA+ABATARgLABQGADAAQY4oDchJ5OAdADICWAdgI40GED6pAXkKkRpAGZQIAW1IwANjCkw2AF1LtSbGAHdSrAM4qAPOxUA+UgG9SOADSks9gMykAvsNEK18xcpUBJVRgoFhkBWQUlVVIAalIULA8RL1IAQzYIFQALYIBRSL9AlWDQwQjfVSKSmTh4xIJRUn0fMG8C1W5CQjEWBXokcixSXIRitn0WCHHCSyTSAHpF0iN9AFcpKVSoAE8zOcX50gBxGBV9NNI5FkNSCDFllW2ABxgnrdSpWHu2D5gBACIACoAJRgazkKn+8wsPiiZwY+yWRnmaw2W12cwwLgoAwMxhBYNWEIsAGUWvj7ATweYABRzRrZa56a74mHtM62emiDAAViMgMpoOpFmaCla0DmAEo5rMGo1RAA3LakWDUspaXR4/lCom02F+fQMADCEFWqkl3HlojE0BgqTAWVINKVUHKcPUbDdBulcqtjVVuoYAEEACYhmmimDiqA0/Xmi1ckQYADsKsJEMtoncBFcQA===)
